bot/cogs/chain_cog.py

```diff
@@ -0,0 +1,104 @@
+import logging
+
+import discord
+import discord.ext.commands as commands
+from copy import copy
+
+import bot.extensions as ext
+
+log = logging.getLogger(__name__)
+
+class ChainCog(commands.Cog):
+    """This cog handles the chaining of commands"""
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.resultqueue = []
+
+    @ext.command()
+    @ext.chainable_output(False)
+    @ext.long_help('Chains multiple chainable commands together')
+    @ext.short_help('Chains commands')
+    @ext.example('chain command1 command2 hello')
+    async def chain(self, ctx, *, text: str):
+        prefix = str(await self.bot.current_prefix(ctx))
+        # Make a fake ctx to intercept values passed to send
+        fakectx = copy(ctx)
```

Comment:
what is the copy for

Reply:
If the copy isnt used, fakectx becomes a reference to ctx, meaning that overriding fakectx.send() overrides ctx.send